examples/cpp/sslputget/CMakeLists.txt.in

```diff
@@ -47,14 +47,16 @@ file(GLOB SSL_CERTIFICATES
 
 file(INSTALL ${SSL_CERTIFICATES} DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
+# This is needed on Linux to force libcryptoImpl.so to be linked in
+if ("Linux" STREQUAL ${CMAKE_SYSTEM_NAME} )
+  set_target_properties(${PROJECT_NAME} PROPERTIES
+    LINK_WHAT_YOU_USE TRUE)
```

Comment:
Why is this a Linux only thing. The current implementation of this property is only applicable to the GNU linker, which is available on more than just Linux and isn't the only linker on Linux. Just set it regardless of OS. It should have no effect where not applicable.

Reply:
Originally I had this property set for all platforms. However, on Mac I get this error:
ld: unknown option: --no-as-needed
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Reply:
Open an issue with CMake about this. Add a comment with a link to the CMake issue you created.

Reply:
Cmake ticket created: https://gitlab.kitware.com/cmake/cmake/issues/20174

Reply:
Added a comment to our CmakeLists.txt about this ticket.